text/0000-api-evolution-transitive-dependencies.md

```diff
@@ -0,0 +1,221 @@
+library, either directly or transitively.
+
+The term "sibling crate" is a crate depended on by an downstream crate.
+
+For the purposes of this RFC, we need to break dependencies into two groups:
+public and private.
+
+A dependency is defined as public when any item provided by that crate is used
+publicly. A dependency is private when no items are used publicly.
+
+Public usage means different things based on the type of item.
+
+For structs, enums, and type aliases public usage includes:
+
+- Using that type as the type of a public field of a public struct or enum
+- Using that type in the signature of a public function or method
+- Implementing a public trait for that type
+- Constraining the impl of a public trait based on that type
+- Referencing that type in a public type alias
+
+For structs, enums, and type aliases public usage does not include:
+
+- Using that type in a private field of a struct or enum
+- Using that type in a position that accepts a generic argument where none of
+  the bounds are provided by your library
+
+For traits public usage includes:
+
+- Implementing that trait for any public struct or enum
+- Using that trait as a supertrait of any public trait
+- Using that trait as a bound on a type parameter of any public item
+
+For all items public usage does not include:
+
+- Publicly re-exporting that item in a public module
+  - Changes to versions are always minor changes with regards to re-exports.
+    This is expanded on below.
+
+## Policy by cargo feature
+
+With these terms defined, we can now look at what is a major vs minor change.
+
+### Major change: modifying the version range of a public dependency to anything that is not a superset of the previous version range
+
+This applies even if the new version range is fully semver compatible with the
+previous one. For example, changing from `=1.0.0` to `=1.0.5` is still
+considered a major change.
+
+Unlike some package managers, Cargo allows multiple versions of the same
+transitive dependency to be used. However, items exported by that crate will be
+considered different types if they come from different versions. Therefore,
+if a version of a dependency is changed from `2.0.0` to `3.0.0`, any place where
+items from that dependency appear publicly could be treated as changing from
+taking `A` to taking `B`, which is a major change under [RFC #1105][1105].
+
+It should be noted that this cannot be avoided by relying on a trait instead of
+a concrete type. If we treat a struct from two different versions as concrete
+types `A` and `B`, changing the version of a dependency can be seen as removing
+an impl of that trait from `A`, and adding it to `B`. Removing an existing trait
+is not explicitly mentioned as a major change by [RFC #1105][1105], but this RFC
+considers it to be so.
+
+As an extension of this, it should be noted that it is impossible to update a
+dependency which contains a breaking change that affects your library, without
```

Comment:
That's not correct. Consider if you only use items of a crate publicly that were not broken. Then your users won't notice anything.